### PR TITLE
configured spring boot, created dao,service,entity layers

### PR DESCRIPTION
**Pull request by orest40th**

### CODE REVIEW

File: src/main/java/com/mate/academy/demo/model/Book.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
```
Comment (Elena-Bruyako) [inline src/main/java/com/mate/academy/demo/model/Book.java]: don't use `@Data` for entites, use `@Getter and @Setter `
Also, add `@Table `

File: src/main/java/com/mate/academy/demo/repository/BookDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            session.persist(book);
+            transaction.commit();
+
+            return book;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+
+            throw new RuntimeException("Cannot insert entity into DB: " + book, e);
```
Comment (Elena-Bruyako) [inline src/main/java/com/mate/academy/demo/repository/BookDaoImpl.java]:
```suggestion
            throw new DataProcessingException("Cannot insert entity into DB: " + book, e);
```

File: src/main/java/com/mate/academy/demo/repository/BookDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("from Book", Book.class).getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Cannot fetch all entities from DB: "
```
Comment (Elena-Bruyako) [inline src/main/java/com/mate/academy/demo/repository/BookDaoImpl.java]:
```suggestion
            throw new DataProcessingException("Cannot fetch all books from DB: "
```

File: src/main/java/com/mate/academy/demo/repository/BookDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                    + Book.class.getSimpleName(), e);
```
Comment (Elena-Bruyako) [inline src/main/java/com/mate/academy/demo/repository/BookDaoImpl.java]:
```suggestion
                    , e);
```

File: src/main/java/com/mate/academy/demo/service/BookServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
+package com.mate.academy.demo.service;
+
+import com.mate.academy.demo.model.Book;
+import com.mate.academy.demo.repository.BookRepository;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
```
Comment (Elena-Bruyako) [inline src/main/java/com/mate/academy/demo/service/BookServiceImpl.java:13]:
```suggestion
@Service
@RequiredArgsConstructor
```

File: src/main/java/com/mate/academy/demo/service/BookServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+
+    @Autowired
+    public BookServiceImpl(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/com/mate/academy/demo/service/BookServiceImpl.java]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/com/mate/academy/demo/repository/BookDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+    @Autowired
+    public BookDaoImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public Book save(Book book) {
+        Transaction transaction = null;
+        try (Session session = sessionFactory.openSession()) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/mate/academy/demo/repository/BookDaoImpl.java]: don't use try with resources and transaction, session will close before catch block

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

**Comment by orest40th** [comment]
i also added a tostring method to see what i fetch in my run()

**Review by liliia-ponomarenko — APPROVED** [review]
Good job! 👍

**Pull request merged**